cmd/linkwarp/main.go

```diff
@@ -2,47 +2,135 @@ package main
 
 import (
 	"context"
-	"fmt"
+	"io"
+	"log"
 	"os"
 	"path/filepath"
 
 	"github.com/warpfork/linkwarp"
+
+	"github.com/jawher/mow.cli"
+)
+
+type ExitCode int
+
+const (
+	ExitCodeOkay ExitCode = iota
+	ExitCodeUnknownError
+	ExitCodePanic
+	ExitCodeError    ExitCode = 25
+	ExitCodeMetanoia ExitCode = 26
+	ExitCodeUsage    ExitCode = 27
 )
 
+// Go up three, because we want to exit "bin/", exit (presumably) "linkwarp*/", exit (presumably) "apps/", and then enter "apps/" and "bin/" again explicitly from there.
+// This results in linkwarp DTRT'ing if installed in the canonical sort of location warpsys conventions would have you put it in, and in all links still being relative.
+const (
+	DefaultAppsPath = "../../../apps/"
+	DefaultBinPath  = "../../../bin/"
+)
+
+// 1 arg form: linkwarp
+// 2 arg form: linkwarp [root]
+// 3 arg form: linkewarp [search root] [bin root]
 func main() {
-	var searchRoot, binRoot string
-	switch len(os.Args) {
-	case 1:
+	ctx := &ActionContext{
+		Context: context.Background(),
+	}
+	app := cli.App("linkwarp", "does things!\nWith no arguments will start at the path of the linkwarp executable")
```

Comment:
What is this even? I didn't feel like thinking about this very hard. Suggestions welcome.